### DISCUSSION TITLE
Libraries

### DISCUSSION THREAD

**Discussion by crista** [dafny-lang/dafny#3979] [General]
Hi,

Is there a separate project for present and future libraries for Dafny? If someone would like to contribute in library-land, what's the best starting point?

Also, is there an IRC or Slack for Dafny, for asking these kinds of questions? (I didn't find it under "Contributing")

Thanks.

**Comment by atomb** [comment]
The closest thing currently exists is probably the [libraries](https://github.com/dafny-lang/libraries) repository.

And we don't currently have a Slack or IRC channel. Any thoughts on what sort of discussion medium folks would find most useful would be very welcome!

**Reply by crista** [reply]
It's always good to have a messaging channel, where ppl can go and talk to you more or less in real time. I'm an IRC fan, but the kids seem to use Slack or Discord.

**Comment by davidcok** [comment]
And I'll add that the intended future library is based on the src/dafny folder, which is a reorganized and renamed version of what is in src.

**Comment by crista** [comment]
Thanks!
The other project doesn't seem to have a Discussion tab, so let me ask it here:
Which libraries are the most needed? Is string processing of interest? Math?
Is the current library written for 4.0 or does it need updating? 
Is there a list of "things we'd love help with"?

**Comment by keyboardDrummer** [comment]
> Also, is there an IRC or Slack for Dafny, for asking these kinds of questions? (I didn't find it under "Contributing")

Have you looked at the gitter? https://app.gitter.im/#/room/#dafny-lang_community:gitter.im
